pyClient/commands/utils.py

```diff
@@ -185,8 +186,8 @@ def _do_sync() -> int:
                 # then attempt to decrypt and validate notes intended for us.
 
                 for out_ev in mix_result.output_events:
-                    merkle_tree.set_entry(
-                        out_ev.commitment_address, out_ev.commitment)
+                    merkle_tree.insert(out_ev.commitment)
```

Comment:
Seems like you no longer use the `commAddr` entry of the `LogCommitment` event. I'd be tempted to keep the "set_entry" function to insert a commitment at a given index in the tree; but if we remove the function (as you did in this PR); then is it worth keeping the `commAddr` entry in the event?

Reply:
Yeah, good point.  I guess it's been redundant ever since the client-side merkle tree was put in place. I looked quickly into removing it.  It shouldn't be diffcult, but it's not quite trivial since the wallet relies on it (as it doesn't have access to the Merkle tree at the moment).  So, it's a small refactor, but I'd lean towards creating an issue and dealing with it in a separate small PR.

Reply:
Created issue #162

Reply:
> I guess it's been redundant ever since the client-side merkle tree was put in place

Seems like it could still be relevant even with the "client-side Merkle tree", since you may want to be able to insert a leaf at a given index depending how you scan the chain. That being said, since the "sync" procedure scans all blocks sequentially, the index of the commitment indeed becomes redundant (each commitments are added to the tree to the "next index" - which is what you do here with the "insert" function)

Reply:
> I'd lean towards creating an issue and dealing with it in a separate small PR.

That works, thanks

Reply:
> Seems like it could still be relevant even with the "client-side Merkle tree", since you may want to be able to insert a leaf at a given index depending how you scan the chain.

Just to clarify, by "redundant", I was only referring to the `commAddr` event.
Regarding the method to insert leaves at arbitrary indices, I'd favour keeping it out for now.  It adds a bit of complexity to the implementation and would be unused code.

Reply:
Yes, that works